### PR TITLE
DOC: Add note about not contain blank in AZ

### LINKED ISSUE #1379: Node registration fails if nova AZ value has ' '
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

inspired by 

https://github.com/kubernetes/cloud-provider-openstack/pull/1364

```
Unable to register node "ocpcic-x45pc-master-0" with API server: Node "ocpcic-x45pc-master-0" is invalid: metadata.labels: Invalid value: "Default Group": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')
```

we suffered above issue with similar pattern, so I think we might consider do same thing to #1364 
**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version:
- OpenStack version:
- Others:

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
fixes #1379

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jichenjc** [comment]
@lingxiankong @ramineni not sure this is the right place to add ,please let me know your comments

**Comment by lingxiankong** [comment]
Is it possible we also convert blank to `-`? The added description is not useful for the cloud admin as changing az name could involve lots of work that should be avoidable.

**Comment by jichenjc** [comment]
> 
> 
> Is it possible we also convert blank to `-`? The added description is not useful for the cloud admin as changing az name could involve lots of work that should be avoidable.

pleaes check https://github.com/kubernetes/cloud-provider-openstack/pull/1380
maybe you can comment on that issue so that we can combine the discussion together

**Comment by lingxiankong** [comment]
> > Is it possible we also convert blank to `-`? The added description is not useful for the cloud admin as changing az name could involve lots of work that should be avoidable.
> 
> pleaes check #1380
> maybe you can comment on that issue so that we can combine the discussion together

OK, it's marked as WIP, so I assumed you are still working on your side rather than ready for review and discussion.

**Comment by jichenjc** [comment]
> > > Is it possible we also convert blank to `-`? The added description is not useful for the cloud admin as changing az name could involve lots of work that should be avoidable.
> > 
> > 
> > pleaes check #1380
> > maybe you can comment on that issue so that we can combine the discussion together
> 
> OK, it's marked as WIP, so I assumed you are still working on your side rather than ready for review and discussion.

yes, it's WIP because I hope to get some comments :) please help to comment on that, thanks

File: docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ For more information about cloud-controller-manager, please see:
 
 **NOTE: Now, the openstack-cloud-controller-manager implementation is based on OpenStack Octavia, Neutron-LBaaS has been deprecated in OpenStack since Queens release and no longer maintained in openstack-cloud-controller-manager. So make sure to use Octavia if upgrade to the latest openstack-cloud-controller-manager docker image.**
 
+**NOTE: `topology.kubernetes.io/zone` is used to label node and its value comes from availability zone of the node, according to [label spec](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) it does not support blank (' ') but OpenStack availability zone supports blank. So your OpenStack availability zone must not contain blank otherwise it will lead to node register issue, see [#1379](https://github.com/kubernetes/cloud-provider-openstack/issues/1379) for further information.**
```
Comment (ramineni) [inline docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md:34]: @jichenjc I would prefer adding in separate Limitations section below. And also , please be explicit , node registration fails if node AZ contains blank in its name.

Comment (jichenjc) [reply docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md:34]: ok, will do

File: docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md [hunk 2]
```diff
@@ -241,3 +243,9 @@ Refer to [Exposing applications using services of LoadBalancer type](./expose-ap
 ## Metrics
 
 Refer to [Metrics for openstack-cloud-controller-manager](../metrics.md)
+
+## Limitation
+
+### OpenStack availability zone must not contain blank
+
+`topology.kubernetes.io/zone` is used to label node and its value comes from availability zone of the node, according to [label spec](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) it does not support blank (' ') but OpenStack availability zone supports blank. So your OpenStack availability zone must not contain blank otherwise it will lead to node that belongs to this availability zone register failure, see [#1379](https://github.com/kubernetes/cloud-provider-openstack/issues/1379) for further information.**
```
Comment (lingxiankong) [inline docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md:251]: Please remove the extra `**`

Comment (jichenjc) [reply docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md]: @lingxiankong updated, thanks

**Pull request merged**